lib/pbench/agent/results.py

```diff
@@ -144,58 +132,66 @@ def make_md5sum(self, tarball):
                     self.logger.error(
                         "error removing failed tar ball MD5, {}", tarball_md5
                     )
-            sys.exit(1)
+            raise RuntimeError("Error in making tarball.md5", exc)
 
 
 class CopyResultTb:
     chunk_size = 4096
 
     def __init__(self, controller, tarball, config, logger):
         self.tarball = Path(tarball)
-        if not self.tarball.exists():
-            logger.error("tarball does not exist, '{}'", tarball)
-            sys.exit(1)
         self.tarball_md5 = Path(f"{tarball}.md5")
-        if not self.tarball_md5.exists():
-            logger.error("tarball's .md5 does not exist, '{}'", self.tarball_md5)
-            sys.exit(1)
         self.logger = logger
-        server_rest_url = config.results.get("server_rest_url")
-        self.upload_url = f"{server_rest_url}/upload/ctrl/{controller}"
+        server_rest_url = config.get("results", "server_rest_url")
+        self.upload_url = f"{server_rest_url}/upload/{controller}/localhost"
+
+    def check_tb_tbmd5(self):
+        if not self.tarball.exists():
+            raise FileNotFoundError("tarball does not exist, '{}'", self.tarball)
+        if not self.tarball_md5.exists():
+            raise FileNotFoundError(
+                "tarball's .md5 does not exist, '{}'", self.tarball_md5
+            )
 
     def read_in_chunks(self, file_object):
         data = file_object.read(self.chunk_size)
         while data:
             yield data
             data = file_object.read(self.chunk_size)
 
-    def copy_result_tb(self):
+    def copy_result_tb(self, token):
         files = [f for f in self.tarball.parent.iterdir() if f.is_file()]
         file_count = len(files)
         if file_count != 2:
-            self.logger.error(
+            raise Exception(
```

Comment:
Can we do better than a generic `Exception`?

Reply:
This still stands out as the only place you're raising just a generic `Exception`.

Reply:
sure

Reply:
At present I will just change it to Runtime Error